wiki/Tutorials/ JupyterHub/JupyterHub.md

```diff
@@ -0,0 +1,190 @@
+JupyterHub is a multi-user plattform for running Jupyter notebooks. Here we will describe how to deploy a JupyterHub platform on a Kubernetes plattform at the Berlin site.
+
+## Setup cloud project
+
+First, the cloud environment must be set up. We assume here that there is a Kubernetes cluster available for your project. See tutorial XXX to set up a Kubernetes cluster via Kubermatic at the Berlin de.NBI site. We will use a specific node (`deploy-node`) to communicate with the Kubernetes cluster. It requires installation of **kubectl** and **helm3** on the node. Depending on the cloud setup, this can be your local computer or a VM within your OpenStack project. Here, we use a VM in the de.NBI cloud based on Ubuntu version 22.10. See [here](https://cloud.denbi.de/wiki/quickstart/) for creating a single VM in your cloud project.
+
+### Kubernetes
+
+1. We need to install **kubectl**. An installation script can downloaded from the official Kubernetes repository.
+
+```
+curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
+sudo apt-get update
+sudo install -o root -g root -m 0755 kubectl /usr/local/bin/kubectl
+kubectl version --client
+```
+
+2. Then we make our kubeconfig (`kubeconfig-XXXXXX`) file available to kubectl. It is good practice, that access to this file is restricted.
+
+```
+mkdir .kube
+cp kubeconfig-XXXXXX .kube/config
+chmod 400 .kube/config
+```
+
+3. To check that cluster is available, we can e.g. display all nodes by typing:
+
+```
+kubectl get nodes
+```
+
+4. We will use a seperate namespace (jhub) to deploy JupyterHub platform:
+
+```
+kubectl create namespace jhub
+```
+
+### Helm
+
+* Helm is required to install JupyterHub. We install Helm version 3 on the `deploy-node`:
+
+```bash
+curl -fsSL -o get_helm.sh https://raw.githubusercontent.com/helm/helm/main/scripts/get-helm-3
+chmod 700 get_helm.sh
+./get_helm.sh
+helm list -A
+```
+
+* We can now add JupyterHub helm charts:
+
+```console
+helm repo add jupyterhub https://jupyterhub.github.io/helm-chart/
+helm repo update
+```
+
+### shares
+
+The JupyerHub in the example will include an extra storage that can be accessed by all users. This can be usefull to share files with all user. Here we use NFS, since this can be easily installed via OpenStack. To create an NFS share in the de.NBI node see [here](https://cloud.denbi.de/wiki/Compute_Center/Berlin/#create-a-nfs-share). If you use a different cloud setup, other storage options can also be integrated. See here (k8s website) to define specific shares. In the example below a 1000GB NFS share is created. To use NFS in the de.NBI cloud the two fields `server` and `path` have to be filled in. The content below needs to be stored as `data-pv.yaml` on the deploy-node.
```

Comment:
```suggestion
The JupyerHub in the example will include an extra storage that can be accessed by all users. This can be useful to share files with all user. Here we use NFS, since this can be easily installed via OpenStack. To create an NFS share in the de.NBI node see [here](https://cloud.denbi.de/wiki/Compute_Center/Berlin/#create-a-nfs-share). If you use a different cloud setup, other storage options can also be integrated. See here (k8s website) to define specific shares. In the example below a 1000GB NFS share is created. To use NFS in the de.NBI cloud the two fields `server` and `path` have to be filled in. The content below needs to be stored as `data-pv.yaml` on the deploy-node.
```

> here (k8s website)

reference missing